snuba/admin/clickhouse/predefined_system_queries.py

```diff
@@ -0,0 +1,104 @@
+from dataclasses import dataclass
+from typing import Dict, Optional, Sequence, Type
+
+
+class _QueryRegistry:
+    """Keep a mapping of SystemQueries to their names"""
+
+    def __init__(self) -> None:
+        self.__mapping: Dict[str, Type["SystemQuery"]] = {}
```

Comment:
```suggestion
        self.__mapping: Dict[str, Type[SystemQuery]] = {}
```

You will need to `from __future__ import annotations` for our version of python